aws-amplify/amplify-js
# feat: (@aws-amplify/ui-components) TOTP / Select MFA Type Components

_Issue #, if available:_
[Delivers: #169738491]

_Description of changes:_
In order to reach feature parity, this work is incorporating the TOTP component and the Select MFA Type component (UC - Utility Component). On the UC, there is an action item to combine the TOTP component with the Select MFA Type. This will need to be done after* the toast component has been completed.

* Unit tests
* Storybook integrations
* Components from React to WC Stencil
* TOTP Example added

Steps to test TOTP: 
1. Use internal `aws-exports` file, ping me if need be
3. Go through the `Amplify Authenticator` under this Pull Request and follow steps provided offline.
3a. QR Code will appear to scan. You can use `Google Authenticator`, link for [iOS](https://apps.apple.com/us/app/google-authenticator/id388497605) or [Android](https://play.google.com/store/apps/details?id=com.google.android.apps.authenticator2&hl=en_US), scan the QR Code and follow the prompts on `Google Authenticator` to receive the number needed to finish the TOTP Authentication process. 
3. Once you have successfully configured TOTP, sign out of the application and re-sign in. After the initial log in, you will see a component with `Confirm TOTP code`. Using the `Google Authenticator` application, find the user and type in the code generated.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
